src/azure-cli/azure/cli/command_modules/role/_params.py

```diff
@@ -234,7 +234,9 @@ def load_arguments(self, _):
         c.argument('identifier_uri', help='graph application identifier, must be in uri format')
         c.argument('spn', help='service principal name')
         c.argument('upn', help='user principal name, e.g. john.doe@contoso.com')
-        c.argument('query_filter', options_list=['--filter'], help='OData filter, e.g. --filter "displayname eq \'test\' and servicePrincipalType eq \'Application\'"')
+        c.argument('query_filter', options_list=['--filter'],
+                   help='The $filter OData query parameter to retrieve a subset of a collection. '
+                        'For more information, see https://learn.microsoft.com/graph/filter-query-parameter')
```

Comment:
This topic is quite similar to the one we are having on `--query`: https://github.com/Azure/azure-cli/pull/24633#discussion_r1069071670, https://github.com/Azure/azure-cli/pull/24332#discussion_r1003912453. That is, whether we want to provide concrete examples for generic arguments in each command.